src/File.php

```diff
@@ -423,10 +431,26 @@ public function canEdit($member = null)
         }
 
         // Delegate to parent if inheriting permissions
-        if ($this->CanEditType === 'Inherit' && $this->ParentID) {
+        if ($this->CanEditType === InheritedPermissions::INHERIT && $this->ParentID) {
             return $this->getPermissionChecker()->canEdit($this->ParentID, $member);
         }
 
+        // Specific user groups can edit this file
+        if ($this->CanEditType === InheritedPermissions::ONLY_THESE_USERS) {
+            if (!$member) {
+                return false;
+            }
+            return $member->inGroups($this->EditorGroups());
+        }
+
+        // Specific users can edit this file
+        if ($this->CanEditType === InheritedPermissions::ONLY_THESE_MEMBERS) {
+            if (!$member) {
+                return false;
+            }
+            return \in_array($member->ID, $this->EditorMembers()->column('ID'), true);
+        }
+
```

Comment:
These shouldn't be necessary - the permission checker canedit call below should cover both of these scenarios.

Reply:
Just tested locally - can confirm that this chunk of code isn't necessary.
Note that because inherited permissions is cached, you may need to either flush or log out and back in again to see the effect of permissions changes.